features/FEATURE_BLE/ble/GattCharacteristic.h

```diff
@@ -1371,13 +1406,19 @@ class GattCharacteristic {
         bool hasVariableLen = true
     ) : _valueAttribute(uuid, valuePtr, len, maxLen, hasVariableLen),
         _properties(props),
-        _requiredSecurity(SecurityManager::SECURITY_MODE_ENCRYPTION_OPEN_LINK),
         _descriptors(descriptors),
         _descriptorCount(numDescriptors),
-        enabledReadAuthorization(false),
-        enabledWriteAuthorization(false),
         readAuthorizationCallback(),
-        writeAuthorizationCallback() {
+        writeAuthorizationCallback(),
+        _update_security(SecurityRequirement_t::NONE) {
+        _valueAttribute.allowWrite(isWritable(_properties));
+        _valueAttribute.allowWrite(isReadable(_properties));
+
+        if (_properties & BLE_GATT_CHAR_PROPERTIES_AUTHENTICATED_SIGNED_WRITES) {
+            _valueAttribute.setWriteSecurityRequirement(
```

Comment:
needs a comment explaining why this translation was chosen

Reply:
Signed writes requires at least Security Mode 2 level 1 which requires at least an unauthenticated CSRK so if signed writes are set then the lowest security requirement is unauthenticated.